logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -25,6 +26,7 @@ def initialize
       # This hash has only one dimension
       # and allow fast retrieval of the metrics
       @fast_lookup = Concurrent::Map.new
+      @structured_lookup_mutex = Mutex.new
```

Comment:
I tend to like to add some documentation close to Mutex declarations to explain their need, what do you think?